src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1713,7 +1713,7 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
 
                 AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(
                     WellKnownMember.System_Reflection_DefaultMemberAttribute__ctor,
-                    ImmutableArray.Create(defaultMemberNameConstant)));
+                    ImmutableArray.Create(defaultMemberNameConstant), isOptionalUse: true));
```

Comment:
I guess my question is, why is this ok to not emit?

Reply:
It is not ok, but this is what we were doing for many years. If we start reporting an error (like VB), we are going to break runtime build, at least. I don't think this is worth it. The scenario is very uncommon.

Reply:
My feeling is, this is a test-only component of NativeAOT that is fairly new. It seems likely to me that it should be including this attribute, and not having it could end up masking a bug. /cc @agocke

Reply:
What’s the actual proposal? That the compiler emits an error for this case and the runtime fixes the test code?

I’m fine with that.

Is it possible that anyone could hit the error without hitting the assert? If so the question really is what happens if you get a bug from windows or office about the new error. Will you change the compiler or tell them to fix their code? Whatever is the answer to that question is the path I would take.

Reply:
> That the compiler emits an error for this case and the runtime fixes the test code?

That's what I would do, yes.

Reply:
> Is it possible that anyone could hit the error without hitting the assert?

Sure. The runtime does right now 🙂.